### PR TITLE
Changes to publish entire interaction session data to ipfs and ipfs-c…

### LINKED ISSUE #11: Manage entire TEXA session data on collaborative IPFS clusters
[Collaborative IPFS clusters](https://cluster.ipfs.io/documentation/collaborative/) can be very helpful for communities to manage public information that is "good for all".

I believe that the disclosure of pre-conditions and interactions made with an AI model during critical testing is as important as the result itself.

FYI, TEXA already dumps the result data to the public IPFS network. Check out the README for more information. Further, what would be nice, is to collect the interaction data from the interrogation web form and dump it as part of the session data, along with the results.

We can use collaborative IPFS clusters to manage the data and allow everyone to access it without paving way for data corruption/manipulation.

Moreover, the session data could also help AI researchers to understand more about the reason behind a biased response or non-responsive behavior from a model under test.

### PR DESCRIPTION
**Pull request by pankaj-nayak**
This PR resolves #11 

This change includes additional interaction session data between AiModel and Interrogator and its corresponding quantum score being published with overall result to public ipfs network and ipfs-cluster.

Unit testing Done : 

VISIT: https://explore.ipld.io/#/explore/bafyreidcgddav3js65stodcqzluxmvypykxgtyluy5eaj4bdkime6sqs5y
for interrogation results. This link is public and the result data is written to the public IPFS network!

VISIT: https://kovan.etherscan.io/tx/0xb989814ab79255b4b59f8bef00f1fd5fef9582360971627b1a4759a1b3319cc5
 for transaction details. This link is public and written to the Ethereum Kovan tesnet!
——————————————————————————————————————————————————

VISIT: https://explore.ipld.io/#/explore/bafyreih4gwhj3yitfwqhwbctcdcv2y23u7rwdv4a3khyiseumubnbm7qk4
for interrogation results. This link is public and the result data is written to the public IPFS network!

VISIT: https://kovan.etherscan.io/tx/0x476beccf8e8cd0a53e62ee64750f8dc11172096509d350fb05e8fd777c3239fa
for transaction details. This link is public and written to the Ethereum Kovan tesnet!

---
This change is [Reviewable](https://reviewable.io/reviews/texaproject/texa/12)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,4 +7,6 @@ require (
 	github.com/TexaProject/texajson v0.0.4
 	github.com/TexaProject/texalib v0.0.0-20170504075604-c488f5b05455
 	github.com/ethereum/go-ethereum v1.9.15
+	github.com/ipfs/go-cid v0.0.7
```
Comment (0zAND1z) [inline go.mod]: Did you run `go mod clean` before making the push?

File: main.go [hunk 2]
```diff
@@ -341,6 +364,13 @@ func getSlabJSON(w http.ResponseWriter, r *http.Request) {
 }
 
 func main() {
+
+	host := flag.String("host", "localhost", "identifier for ipfs and ipfs cluster")
+	port := flag.String("port", "9094", "Source port number")
```
Comment (0zAND1z) [inline main.go]: Shouldn't the description be "Port number of ipfs-cluster" instead?

File: go.mod [hunk 3]
```diff
@@ -7,4 +7,6 @@ require (
 	github.com/TexaProject/texajson v0.0.4
```
Comment (0zAND1z) [inline go.mod]: After the texajson PR, I have rolled out a new release. So please make sure that your `go.mod` points to `v0.0.5`. :)

File: www/index.html [hunk 4]
```diff
@@ -17,6 +17,10 @@
     SlabNum = "";
     while (SlabNum == "" || SlabNum == null || SlabNum == 0) {
       SlabNum = prompt("Enter number of Slabs:", "");
+      if (isNaN(SlabNum) == true) {
```
Comment (0zAND1z) [inline www/index.html:20]: Can we do similar form/input validations for all inputs?

Only if it's not too much to ask (optional).

**Review by 0zAND1z — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (pankaj-nayak) [reply go.mod]: no. Done now.

File: main.go (hunk 2 above)
Comment (pankaj-nayak) [reply main.go]: corrected.Done.

File: go.mod (hunk 3 above)
Comment (pankaj-nayak) [reply go.mod]: sure :)

File: www/index.html (hunk 4 above)
Comment (pankaj-nayak) [reply www/index.html:20]: we have following input types : 
1. AI NAME : characters and integers are accepted. 
2. IntName : characters and integers are accepted. 
3. Slab Count : Only Numbers are accepted [Check is there.]
4. Slab Name : characters and integers are accepted. 
5. Justification : characters and integers are accepted. 
6. Transactions : characters and integers are accepted.
7. Quantum Score : Only 0 and 1 are accepted [Check is there]

I did't put check for remaining types thinking - input could be of any kind, for example : Slab name could  be 001, 002 for a particular AI model . For justification and transaction user is free to use both numbers and strings. 

Kindly let me know any case which i might be missing ,i will take care of it.

Comment (0zAND1z) [reply www/index.html:20]: Yes @pankaj-nayak, it could be an alphanumeric string.

**Review by 0zAND1z — APPROVED** [review]

**Pull request merged**